### ISSUE TITLE
Make root and svg read only

### ISSUE THREAD

**Issue by kum-deepak** [dc-js/dc.js#1737]
The current API allows these to be updated. The documentation warns against that. Charts are lilely to be messed up if someone does set these. We should make these read-only with backward compatibility.

@gordonwoodhull, waiting for your views.

**Comment by gordonwoodhull** [comment]
I disagree. We don't protect users from themselves in this library. It's a leaky abstraction by design.

**Comment by kum-deepak** [comment]
Since we moved to ES6, accessing any property is possible, which the current version will continue to allow. Through documented API, I think that we should allow read-only access to these properties.
